test/torch/test_federated_learning.py

```diff
@@ -88,3 +88,24 @@ def test_toy_federated_learning(self):
 
                 # 6) print our progress
                 print(loss.get())  # NEW) slight edit... need to call .get() on loss
+
+
+def test_lstm(workers):
+    bob, alice = workers["bob"], workers["alice"]
+
+    # 5) change those weights
+    lstm = nn.LSTM(3, 3)
+    inputs = torch.randn(5, 1, 3)
+    hidden = (torch.randn(1, 1, 3), torch.randn(1, 1, 3))  # clean out hidden state
+    out, hidden = lstm(inputs, hidden)
+    assert out.shape == torch.Size([5, 1, 3])
+    lstm = nn.LSTM(3, 3)
+    lstm.send(bob)
+    inputs = torch.randn(5, 1, 3).send(bob)
+    hidden = (
+        torch.randn(1, 1, 3).send(bob),
+        torch.randn(1, 1, 3).send(bob),
+    )  # clean out hidden state
+    # out, hidden = lstm(inputs, hidden)
+    # Test will fully pass once the tensor.dim() method is implemented for remote tensors
+    # assert out.shape == torch.Size([5, 1, 3])
```

Comment:
Would like to try implementing .dim() for remote tensors ? Could be  a cool contribution
You could rely on the .shape attribute which works for remote tensors

Reply:
Would it be sufficient to just hook the .dim() method from the original PyTorch, or what exactly what would need to be done to implement it for remote tensors?

Reply:
If you hook dim() to use .shape, then it will also work for remote tensor: namely instead of returning the shape of the pointer itself it sends the call remotely to get the remote shape

Reply:
Implemented dim() for remote tensors

Reply:
👍